### PR TITLE
Ranges, `getindex`, and `compute_inplace`

### PR DESCRIPTION
**Pull request by calebwin**
This PR introduces several new features!

1. Ranges for collecting Julia ranges into Banyan Julia arrays (e.g., `BanyanArrays.collect(1:10:10_000_000)`)
2. `getindex` for indexing Banyan arrays (e.g., `array[:, :, [2, 6], 1:10]`)
3. `compute_inplace` for computing some value without returning it to the client side (e.g., `compute_inplace(results)`

We also fixed some bugs in the annotation of `map` and the implementation of `Copy*` PFs.

### CODE REVIEW

**Comment by calebwin** [comment]
WIP because I'm trying to fix a weird internal Julia stack error when compiling `DelayedTask` :weary:

File: Banyan/src/sessions.jl [hunk 1]
```diff
@@ -329,6 +329,10 @@ function get_session_status(session_id::String=get_session_id(); kwargs...)
     configure(; kwargs...)
     filters = Dict("session_id" => session_id)
     response = send_request_get_response(:describe_sessions, Dict{String,Any}("filters"=>filters))
+    if !haskey(response["sessions"], session_id)
```
Comment (cailinw) [inline Banyan/src/sessions.jl:332]: What is this case for? Not sure I understand

File: BanyanArrays/test/mapreduce.jl [hunk 2]
```diff
@@ -251,6 +251,76 @@ end
...
+        part1_str = read(S3Path("s3://$(get_cluster_s3_bucket_name())/test_getindex_and_collect/part1.txt", config=Banyan.get_aws_config()), String)
+        offloaded() do 
+            bucket = readdir("s3")[1]
+            rm("s3/$bucket/test_getindex_and_collect/", recursive=true)
+        end
+        @test part1_str == string(Base.fill(1.0, 100))
+    end
+end
+
+# @testset "Main worker stuck" begin
```
Comment (cailinw) [inline BanyanArrays/test/mapreduce.jl]: Is this a working test? if so, uncomment

**Review by cailinw — CHANGES_REQUESTED** [review]
looking good, thank you!

File: Banyan/src/sessions.jl (hunk 1 above)
Comment (calebwin) [reply Banyan/src/sessions.jl:332]: It was a weird edge-case I encountered where a session waiting for creating to finish suddenly had an error that the result of trying to get its updated session status was that the session ID was not in the result of `describe-sessions`. This handles that edge-case with a warning.

Comment (cailinw) [reply Banyan/src/sessions.jl:332]: Ah, and this is because of a bug in describe_sessions, which will be fixed soon

**Review by cailinw — APPROVED** [review]

**Pull request merged**